manifests/unit.pp

```diff
@@ -22,55 +22,69 @@
   $install_wantedby        = undef,
 ) {
 
-  validate_re($ensure, [ '^present$', '^absent$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
+  include ::systemd
+
+  if is_string($ensure) {
+    validate_re($ensure, [ '^present$', '^absent$' ],
```

Comment:
how about to replace `validate_re` with [validate_legacy](https://github.com/puppetlabs/puppetlabs-stdlib#validate_legacy)?

Reply:
Good suggestion. However it does not appear it's supported in Puppetv3. 
    Note: This function is compatible only with Puppet 4.4.0 (PE 2016.1) and later.
https://github.com/puppetlabs/puppetlabs-stdlib/tree/5d6cca1d528cd450ec9ac1afcd28c53ea79e3ff4#validate_legacy

Reply:
ah ok. i didn't see the note